### PR TITLE
IDC-1215: CCC uni/bidirectional annotations to Cornerstone tool state

### PR DESCRIPTION
**Pull request by igoroctaviano**
Uses  a new dcmjs Cornerstone MeasurementReport hook to allow defining Crowds Cure Canter matching logic to map a measurement group to a Cornerstone tool class instead of relying on CCC random tracking identifiers e.g. "web annotation" which doesn't work for current SR hydration implementation

- DCMJS PR (already merged): https://github.com/dcmjs-org/dcmjs/pull/201
- Related ticket: https://github.com/OHIF/Viewers/issues/1215
- Datasets for uni directional: https://wiki.cancerimagingarchive.net/display/DOI/Crowds+Cure+Cancer%3A+Data+collected+at+the+RSNA+2017+annual+meeting
- Datasets for bi direictional: https://wiki.cancerimagingarchive.net/display/DOI/Crowds+Cure+Cancer%3A+Data+collected+at+the+RSNA+2018+annual+meeting

### CODE REVIEW

File: platform/viewer/public/config/default.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ window.config = {
     dicomWeb: [
       {
         name: 'DCM4CHEE',
-        wadoUriRoot: 'https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/wado',
-        qidoRoot: 'https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs',
-        wadoRoot: 'https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs',
```
Comment (Punzo) [inline platform/viewer/public/config/default.js:13]: should we set the server one as default?

**Review by Punzo — CHANGES_REQUESTED** [review]
just some questions, since I am not familiar with SR:
1) why is this PR not pushed to https://github.com/OHIF/Viewers as well?
2) Is this a IDC customization since CrowdsCureCancer identifiers are IDC specifics? would this mean that this PR will not work with "general"/"standard" uni/bidirectional annotations?
3) could you please give a brief summary which SR annotations this PR implements, what is already available in the OIHF viewer and what is missing in the context of https://github.com/OHIF/Viewers/issues/1215 and http://docs.google.com/document/d/1bR6m7foTCzofoZKeIRN5YreBrkjgMcBfNA7r9wXEGR4/?

File: platform/viewer/public/config/default.js (hunk 1 above)
Comment (igoroctaviano) [reply platform/viewer/public/config/default.js:13]: Mistake push, ill rollback, thanks

**Comment by igoroctaviano** [comment]
> just some questions, since I am not familiar with SR:
> 
> 1. why is this PR not pushed to https://github.com/OHIF/Viewers as well?
> 2. Is this a IDC customization since CrowdsCureCancer identifiers are IDC specifics? would this mean that this PR will not work with "general"/"standard" uni/bidirectional annotations?
> 3. could you please give a brief summary which SR annotations this PR implements, what is already available in the OIHF viewer and what is missing in the context of [OHIF#1215](https://github.com/OHIF/Viewers/issues/1215) and http://docs.google.com/document/d/1bR6m7foTCzofoZKeIRN5YreBrkjgMcBfNA7r9wXEGR4/?

@Punzo to get you up to speed...

This is IDC specific request. There's an open discussion [here](https://github.com/dcmjs-org/dcmjs/pull/197) and [here](https://github.com/OHIF/Viewers/issues/1215) about a better way to store planar annotations using Clunie's paper. It's not a solid standard but the path we should take, we still need to come up with a solid representation + solid matching that would cover other annotations + including some math checks but this will require more thought and more people. 

This is a temporary solution for the current implementation that uses tracking identifiers as the key to mapping from SR to a Cornerstone annotation. The problem is that this tracking identifier is populated with Cornerstone tool info only when the STOWRS by OHIF. Since these CCC annotations were generated somewhere else + are uni/bi-directional annotations only, i just added a simple and straightforward check for long axis or length (okey for this scope).

**Review by Punzo — APPROVED** [review]

**Comment by Punzo** [comment]
> > just some questions, since I am not familiar with SR:
> > 
> > 1. why is this PR not pushed to https://github.com/OHIF/Viewers as well?
> > 2. Is this a IDC customization since CrowdsCureCancer identifiers are IDC specifics? would this mean that this PR will not work with "general"/"standard" uni/bidirectional annotations?
> > 3. could you please give a brief summary which SR annotations this PR implements, what is already available in the OIHF viewer and what is missing in the context of [OHIF#1215](https://github.com/OHIF/Viewers/issues/1215) and http://docs.google.com/document/d/1bR6m7foTCzofoZKeIRN5YreBrkjgMcBfNA7r9wXEGR4/?
> 
> @Punzo to get you up to speed...
> 
> This is IDC specific request. There's an open discussion [here](https://github.com/dcmjs-org/dcmjs/pull/197) and [here](https://github.com/OHIF/Viewers/issues/1215) about a better way to store planar annotations using Clunie's paper. It's not a solid standard but the path we should take, we still need to come up with a solid representation + solid matching that would cover other annotations + including some math checks but this will require more thought and more people.
> 
> This is a temporary solution for the current implementation that uses tracking identifiers as the key to mapping from SR to a Cornerstone annotation. The problem is that this tracking identifier is populated with Cornerstone tool info only for OHIF STOWRS. Using a tracking identifier to do this mapping is not right but it was a valid POC at the time because it doesn't require smart matching SR properties / scoord to a Cornerstone annotaiton. Since these CCC annotations were generated somewhere else + are uni/bi-directional annotations only, i just added a simple and straightforward check for long axis or length (okey for this scope).

OK, thanks for the exaplanation!

**Comment by fedorov** [comment]
@igoroctaviano can you please:
1) let me know if the sandbox has been updated to include the features you implemented to support CCC annotations 
2) sandbox reports version 4.9.13 - how do I figure out where that is with respect to this PR merged?
3) what is the procedure to get to see those in-plane annotations (in the likely event I get confused)

**Comment by igoroctaviano** [comment]
> @igoroctaviano can you please:
> 
> 1. let me know if the sandbox has been updated to include the features you implemented to support CCC annotations
> 2. sandbox reports version 4.9.13 - how do I figure out where that is with respect to this PR merged?
> 3. what is the procedure to get to see those in-plane annotations (in the likely event I get confused)
> 
> Here's the test study with CCC annotations, which does not work for me: https://idc-sandbox-000.firebaseapp.com/projects/idc-tcia/locations/us-central1/datasets/tcia-idc-datareviewcoordination/dicomStores/CrowdsCure/study/1.3.6.1.4.1.14519.5.2.1.9203.4004.100545948082587796019777812429

I just asked Bill to check that + ask if we have IDC-specific version files to update to keep track. 
The procedure is basically opening the measurement panel and clicking to jump to annotation.

**Pull request merged**